docs/fundamentals/toc.yml

```diff
@@ -653,11 +657,11 @@ items:
         href: ../core/diagnostics/debug-linux-dumps.md
       - name: SOS debugger extension
         href: ../core/diagnostics/sos-debugging-extension.md
-    - name: Logging and tracing
+    - name: Performance tracing
```

Comment:
We need to move ILogger out as its the #1 thing that users will use. 
Then step back and ask yourself "When are users going to use EventSource or DiagnosticSource?" I would suggest that most of that is going to be for diagnosing performance.

Reply:
> We need to move ILogger out as its the https://github.com/dotnet/docs/pull/1 thing that users will use.

If you want to make ILogger extra prominent how about:
```
- ILogger
- Logging and Tracing
  - ILogger 
  - EventSource
    - ...
    - Well-known event providers
    - EventPipe
  - DiagnosticSource
```

I don't think making logger prominent requires anything else to be re-arranged.


> Then step back and ask yourself "When are users going to use EventSource or DiagnosticSource?" I would suggest that most of that is going to be for diagnosing performance.

If a customer asked me I'd answer:
EventSource - 60% diagnosing performance, 20% functional logging, 20% generic out-of-proc transport for other types of telemetry. Prominent examples of its use for functional logging are OTel SDK and Azure SDK.
DiagnosticSource - 60% APM/distributed tracing, 20% custom in-proc diagnostic tools, 20% performance tracing (when combined with DiagnosticSourceEventSource)

I think characterizing these tools as being primarily for "performance tracing" is simplified for EventSource (I acknowledge there are times where simple is good) and wrong for DiagnosticSource. Saying they are "for diagnosing performance" is a somewhat better fit, but that is a much broader characterization than "performance tracing" which would also include technologies like metrics and distributed tracing too.

Over time I think we are going to see increasing scenario crossover between ILogger and EventSource, not less. For example I'd like to change dotnet-trace so that you can specify ILogger categories as providers and have options for text or binary formats. OpenTelemetry is trying to make EventSource be another API you can use to produce the logging signal. I expect creating a view where logging and performance tracing are treated as very different and APIs should only be thought of as doing one or the other will be an increasingly strained organizational principle over time.

Reply:
I think the amount of direct customer usage of EventSource in their code, either emitting or consuming events is going to be minimal - it's going to pretty much be relegated to us as part of the framework/runtime. The interaction that customers will have with EventSource is via a trace, which is most likely to be performance related.